harness/determined/cli/trial.py

```diff
@@ -125,71 +125,72 @@ def trial_logs(args: Namespace) -> None:
         timestamp_after=args.timestamp_after,
     )
 
+
 @authentication.required
-def output_logs(args: Namespace): 
-    if args.output_dir is None: 
-        raise Exception('Please provide the output directory you would like to direct the logs to.')
-    
-    output_dir_tar = tarfile.open(f'{args.output_dir}.tar.gz', 'w:gz')
-
-    trial_logs_file = write_trial_logs(args, args.trial_id, output_dir_tar)
-    output_dir_tar.add(f'{trial_logs_file}.json')
-    os.remove(f'{trial_logs_file}.json')
-    
-    master_logs_file = write_master_logs(args, output_dir_tar)
-    output_dir_tar.add(f'{master_logs_file}.json')
-    os.remove(f'{master_logs_file}.json')
-
-
-    api_trail_file, api_experiment_file = write_api_call(args, output_dir_tar)
-    output_dir_tar.add(f'{api_trail_file}.json')
-    output_dir_tar.add(f'{api_experiment_file}.json')
-    os.remove(f'{api_trail_file}.json')
-    os.remove(f'{api_experiment_file}.json')
-
-    output_dir_tar.close()
-
-    return f'{args.output_dir}.tar.gz'
-
-    
-
-def write_trial_logs(args,trial_id, output_dir):  #difference between this trial_logs and api call? There doesn't seem to be any. 
-    trial_logs = api.trial_logs(args.master, trial_id)
-    # use trial logs to get the experiment id 
-    file_name = 'trial_logs'
-    trial_logs_list = []
-    for log in trial_logs: 
-        trial_logs_list.append(log)
-
-    create_json_file_in_dir(trial_logs_list,file_name, output_dir)
-    return file_name
-
-def write_master_logs(args, output_dir):
-    response = api.get(
-                args.master, "logs"
-            )
-    file_name = 'master_logs'
-    create_json_file_in_dir(response.json(),file_name, output_dir)
-    return file_name
-
-def write_api_call(args, output_dir): 
-    file_name1 = 'api_experiment_call'
-    file_name2 = 'api_trial_call'
+def output_logs(args: Namespace):
+    temp_dir = tempfile.TemporaryDirectory()
+    trial_logs_filepath = write_trial_logs(args, temp_dir)
+
+    master_logs_filepath = write_master_logs(args, temp_dir)
+
+    api_experiment_filepath, api_trail_filepath = write_api_call(args, temp_dir)
+
+    bundle = None
+    if args.output_dir is None:
+        output_dir = os.getcwd()
+        tar = f"det-bundle-trail_{args.trial_id}-{datetime.now().strftime('%Y-%m-%d_%H:%M:%S')}.tar.gz"
+        fullpath = os.path.join(output_dir, tar)
+        bundle = tarfile.open(fullpath, "w:gz")
+
+    else:
+        bundle = tarfile.open(f"{args.output_dir}.tar.gz", "w:gz")
```

Comment:
We don't want whatever the user specifies as the output directory to be the name of the file. For example, if the user says, `det t support-bundle 1234 -o /tmp`, we don't want it to try and write `/tmp.tar.gz`. We want it to do something like `/tmp/det-bundle-trial-1234-20220415T105733.tar.gz`


```
output_dir = os.getcwd()
if args.output_dir is not None:
    output_dir = args.output_dir

filename_printf = "det-bundle-trial-{trial_id}-{datetime}.tar.gz"
tar_filename = filename_printf.format(trial_id=args.trail_id, datetime=datime.now().strftime('%Y%m%dT%H%M%S'))
fullpath = os.path.join(output_dir, tar_filename)
bundle = tarfile.open(fullpath, "w:gz")
```

... I also changed the format of the time a bit. I don't know if Windows will get angry about `:` in paths, we should probably avoid it.